conans/client/graph/graph_manager.py

```diff
@@ -57,15 +56,28 @@ def load_consumer_conanfile(self, conanfile_path, info_folder, output,
                                 deps_info_required=False):
         """loads a conanfile for local flow: source, imports, package, build
         """
-        profile = read_conaninfo_profile(info_folder) or self._client_cache.default_profile
-        cache_settings = self._client_cache.settings.copy()
-        cache_settings.values = profile.settings_values
-        # We are recovering state from captured profile from conaninfo, remove not defined
-        cache_settings.remove_undefined()
-        processed_profile = ProcessedProfile(cache_settings, profile, None)
+        try:
+            graph_info = GraphInfo.load(info_folder)
+            profile = graph_info.profile
+            profile.process_settings(self._client_cache, preprocess=False)
+            # This is the hack of recovering the options from the graph_info
+            profile.options.update(graph_info.options)
+        except Exception:
+            # This is very dirty, should be removed for Conan 2.0 (source() method only)
+            profile = self._client_cache.default_profile
+            profile.process_settings(self._client_cache)
+        processed_profile = ProcessedProfile(profile, None)
         if conanfile_path.endswith(".py"):
             conanfile = self._loader.load_conanfile(conanfile_path, output, consumer=True,
-                                                    local=True, processed_profile=processed_profile)
+                                                    processed_profile=processed_profile)
+            with get_env_context_manager(conanfile, without_python=True):
```

Comment:
Why are we calling the config methods now and not previously?

Reply:
because now we are not doing magic anymore trying to recover the state of the consumer conanfile, but actually computing it, re-evaluating